tracker/dmlc_tracker/ssh.py

```diff
@@ -20,23 +21,67 @@ def sync_dir(local_dir, slave_node, slave_dir):
         slave_node[1], local_dir, remote)
     subprocess.check_call([prog], shell = True)
 
-def get_env(pass_envs):
+def prepare_envs(args):
```

Comment:
Since the logic for prepare_envs seems to be common for ssh and local it would be nice to write a common function in the future which can be used by both modules.

Reply:
All the common work is handled by parse_env_pairs method in opts.py 
Other than that, we have some differences between local and ssh, because in local we copy all environment variables which are set. But for ssh we pass specific ones. There's no redundant code as such here.